### PR TITLE
Implemented Drivetrain class with Talon PID + small other additions

### PR DESCRIPTION
**Pull request by BBScholar**
Focus on drivetrain class

### CODE REVIEW

File: src/main/kotlin/frc/team5499/frc2018Kotlin/Constants.kt [hunk 1]
```diff
@@ -3,24 +3,73 @@ package frc.team5499.frc2018Kotlin
...
+        const val INVERT_TURN_AUX_PIDF = false
+    }
+
+    const val EPSILON = 1E-10
     const val ENCODERS_TICKS_PER_ROTATION = 1028
     const val WHEEL_DIAMETER = 6.0 // inches
     const val WHEEL_CIR = WHEEL_DIAMETER * 2.0
+    const val TRACK_WIDTH = 20.0
+    const val MAX_VELOCITY_SETPOINT = 120.0 // inches per second
+    const val UPDATE_PERIOD: Double = 0.005
```
Comment (andycate) [inline src/main/kotlin/frc/team5499/frc2018Kotlin/Constants.kt]: Why does this have `: Double`

File: src/main/kotlin/frc/team5499/frc2018Kotlin/Robot.kt [hunk 2]
```diff
@@ -2,25 +2,54 @@ package frc.team5499.frc2018Kotlin
...
 class Robot : TimedRobot() {
 
     override fun robotInit() {
-        println("Hello World from Kotlin!")
+        super.setPeriod(Constants.UPDATE_PERIOD)
     }
 
-    override fun robotPeriodic() {}
+    override fun robotPeriodic() {
+        Drivetrain.broadcastToDashboard()
```
Comment (andycate) [inline src/main/kotlin/frc/team5499/frc2018Kotlin/Robot.kt]: Get rid of this. We will add this functionality later, but probably not using broadcast functions like this.

File: src/main/kotlin/frc/team5499/frc2018Kotlin/Robot.kt [hunk 3]
```diff
@@ -2,25 +2,54 @@ package frc.team5499.frc2018Kotlin
...
+        Arm.broadcastToDashboard()
+        Intake.broadcastToDashboard()
```
Comment (andycate) [inline src/main/kotlin/frc/team5499/frc2018Kotlin/Robot.kt]: Again, these two lines aren't necessary

File: src/main/kotlin/frc/team5499/frc2018Kotlin/subsystems/Drivetrain.kt [hunk 4]
```diff
@@ -160,8 +185,30 @@ object Drivetrain : Subsystem() {
 
     private fun configForVelocity() {
         mLeftMaster.apply {
-            configSelectedFeedbackSensor(FeedbackDevice.QuadEncoder, 0, 0)
             inverted = false
```
Comment (andycate) [inline src/main/kotlin/frc/team5499/frc2018Kotlin/subsystems/Drivetrain.kt]: is this variable ever used?

File: src/main/kotlin/frc/team5499/frc2018Kotlin/subsystems/Intake.kt [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    override fun update() {
+    }
+
+    override fun stop() {
+    }
+
+    override fun reset() {
+    }
+
+    override fun broadcastToDashboard() {
```
Comment (andycate) [inline src/main/kotlin/frc/team5499/frc2018Kotlin/subsystems/Intake.kt]: Lowkey should just delete all the broadcastToDashboard functions

File: src/main/kotlin/frc/team5499/frc2018Kotlin/subsystems/Drivetrain.kt (hunk 4 above)
Comment (BBScholar) [reply src/main/kotlin/frc/team5499/frc2018Kotlin/subsystems/Drivetrain.kt]: inverted?

File: src/main/kotlin/frc/team5499/frc2018Kotlin/Constants.kt (hunk 1 above)
Comment (BBScholar) [reply src/main/kotlin/frc/team5499/frc2018Kotlin/Constants.kt]: ill fix that

File: src/main/kotlin/frc/team5499/frc2018Kotlin/subsystems/Intake.kt (hunk 5 above)
Comment (BBScholar) [reply src/main/kotlin/frc/team5499/frc2018Kotlin/subsystems/Intake.kt]: ok

**Pull request merged**